modules/RaptMediaNg/resources/mw.RaptMediaNg.js

```diff
@@ -0,0 +1,508 @@
+/**
+ * The RaptMediaNg plugin integrates the RaptMedia Engine to the Kaltura Player.
+ * RaptMedia adds clickable interactive layer that accompanies your video content and can do things like:
+ * cue or launch different media plays, jump to specific timecode, trigger an event on your webpage and launch a new web page or an app.
+ * This plugin also makes use of accompanying plugin RaptMediaNgScrubber plugin to override the default scrubber behavior to fit a Rapt Media experience.
+ * With RaptMediaNgScrubber plugin the scrubber can interact within the context of a single RaptMedia clip instead of just the entire stitched playlist.
+ * The RaptMediaNg plugin integrates the RaptMedia Engine to the Kaltura Player.
+ * It also makes use of accompanying plugin RaptMediaNgDurationLabel used to override the default player DurationLabel to behave according to the RaptMedia Sequence rather than show the overall playlist duration.
+ */
+(function ( mw, $ ) {
+	"use strict";
+
+	var AbortError = function() {};
+
+	mw.PluginManager.add( 'raptMediaNg', mw.KBaseComponent.extend( {
+
+		defaultConfig: {
+			parent: 'videoHolder'
+		},
+
+		setup: function(){
+			this.initialize();
+			this.addBindings();
+		},
+
+		initialize: function() {
+			this.setConfig('status', 'disabled', true);
+
+			this.setConfig('projectId', undefined, true);
+			this.setConfig('info', undefined, true);
+
+			this.pendingEntryId = null;
+			this.targetEntryId = null;
+		},
+
+		addBindings: function() {
+			var _this = this;
+
+			this.bind('raptMediaNg_doPlay', function(event) {
+				_this.execute({ type: 'player:play' });
+			});
+
+			this.bind('raptMediaNg_doPause', function(event) {
+				_this.execute({ type: 'player:pause' });
+			})
+
+			this.bind('raptMediaNg_doSeek', function(event, time) {
+				_this.execute({ type: 'player:seek', payload: { time: time } });
+			});
+
+			this.bind('raptMediaNg_doJump', function(event, locator) {
+				_this.execute({ type: 'project:jump', payload: { destination: locator } });
+			});
+
+			this.bind('raptMediaNg_doReplay', function(event) {
+				_this.execute({ type: 'project:replay' });
+			});
+
+			this.bind('raptMediaNg_doCommand', function(event, command) {
+				_this.execute(command);
+			});
+
+			this.bind('checkPlayerSourcesEvent', function(event, callback) {
+				_this.playbackCallback = callback;
+			});
+
+			this.bind('KalturaSupport_EntryDataReady', function(event) {
+				// KalturaSupport_EntryDataReady can be called synchronously from a
+				// `checkPlayerSourcesEvent` handler if the required data is already
+				// cached. In that case `_this.playbackCallback` may not be available
+				// synchronously, so we force asynchronous evaluation
+				setTimeout(function() {
+					_this.log('Checking if Entry is an Interactive Video');
+
+					var raptProjectId = _this.readRaptProjectId();
+					if (raptProjectId) {
+						_this.once('raptMediaNg_ready', _this.playbackCallback);
```

Comment:
This is another event designed for external consumption, as such I would consider leaving it with the `raptMedia_` prefix.